web/src/lib/components/alert/alert.tsx

```diff
@@ -0,0 +1,49 @@
+import { h, Component } from 'preact';
+import Icon from '../icon';
+
+interface Props {
+  text: string;
+  active: boolean;
+}
+
+interface State {
+  active: boolean;
+}
+
+export default class Alert extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.onClick = this.onClick.bind(this);
+  }
+
+  state: State = {
+    active: false,
+  };
+
+  componentWillReceiveProps(nextProps: Props) {
+    if (nextProps.active !== this.props.active) {
```

Comment:
by comparing with `this.props` instead of `this.state`, the following behavior can happen:

1. parent component sets banner to visible (`props.active == true`, `state.active == true`)
1. user clicks banner to dismiss it (`props.active == true`, `state.active == false`)
1. parent component re-sets the props of the alert component to `true`
* -> line 25 does not trigger `setState`, banner does not become visible.

this seems like a somewhat unlikely scenario, but if it does happen, it will be difficult to track down.

-> instead, compare `nextProps` to `this.state`, since that reflects the thing you (might) want to change :)